app/models/ability.rb

```diff
@@ -0,0 +1,37 @@
+class Ability
+  include CanCan::Ability
+  def initialize(user)
+    return if user.blank?
+
+    can :destroy, Post do |post|
+      post.user == user || user.role == 'admin'
+    end
+    can :destroy, Comment do |comment|
+      comment.user == user || user.role == 'admin'
+    end
+    # Define abilities for the user here. For example:
+    #
+    #   return unless user.present?
+    #   can :read, :all
```

Comment:
![image](https://user-images.githubusercontent.com/65068771/224120916-fe30439a-c88b-4b3b-8414-2b1f3c28d560.png)


- [ ] kindly uncomment(` can :read, :all`) this to allow the user to access  other pages for now I think the authorization is made only on destroying comments and posts